Fix frontend CI

## Issue description
Frontend CI fails:
```
/home/runner/work/PROTzilla/PROTzilla/frontend/src/components/core/shared/button/button.tsx
Error:   22:12  error  Calling a string's .toString() method does not change the type or value of the string  @typescript-eslint/no-unnecessary-type-conversion

/home/runner/work/PROTzilla/PROTzilla/frontend/src/components/core/shared/icon/icon.tsx
Error:   34:10  error  Calling a string's .toString() method does not change the type or value of the string  @typescript-eslint/no-unnecessary-type-conversion
Error:   35:10  error  Calling a string's .toString() method does not change the type or value of the string  @typescript-eslint/no-unnecessary-type-conversion
Error:   36:10  error  Calling a string's .toString() method does not change the type or value of the string  @typescript-eslint/no-unnecessary-type-conversion

/home/runner/work/PROTzilla/PROTzilla/frontend/src/components/core/shared/input-fields/input-container/input-container.tsx
Error:   30:45  error  Calling a string's .toString() method does not change the type or value of the string  @typescript-eslint/no-unnecessary-type-conversion

/home/runner/work/PROTzilla/PROTzilla/frontend/src/models/sync-engine/rest-adapter.ts
Error:   26:59  error  Passing a string to String() does not change the type or value of the string  @typescript-eslint/no-unnecessary-type-conversion
Error:   47:59  error  Passing a string to String() does not change the type or value of the string  @typescript-eslint/no-unnecessary-type-conversion
Error:   54:59  error  Passing a string to String() does not change the type or value of the string  @typescript-eslint/no-unnecessary-type-conversion

/home/runner/work/PROTzilla/PROTzilla/frontend/src/utils/platform-detection.ts
Error:   16:7  error  Passing a boolean to Boolean() does not change the type or value of the boolean  @typescript-eslint/no-unnecessary-type-conversion

✖ 9 problems (9 errors, 0 warnings)

 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

## Acceptance Criteria
- [x] Frontend CI does not fail anymore

--- a/frontend/src/components/core/shared/button/button.tsx
+++ b/frontend/src/components/core/shared/button/button.tsx
@@ -19,7 +19,7 @@ import {
 const StyledButton = styled.button.withConfig({
   shouldForwardProp: (prop: string) =>
     !["isDisabled", "isActive", "onlyIcon", "onlyText", "isShy", "isSmall"].includes(
-      prop.toString(),
+      prop,
     ),
 })<
   Pick<ButtonProps, "iconRight" | "isDisabled" | "isShy" | "isSmall"> & {

--- a/frontend/src/components/core/shared/icon/icon.tsx
+++ b/frontend/src/components/core/shared/icon/icon.tsx
@@ -30,10 +30,10 @@ export const iconColor =
   `;
 
 const StyledSVG = styled.svg.withConfig({
-  shouldForwardProp: (prop) =>
-    prop.toString() !== "isDisabled" &&
-    prop.toString() !== "color" &&
-    prop.toString() !== "isSmall",
+  shouldForwardProp: (prop: string) =>
+    prop !== "isDisabled" &&
+    prop !== "color" &&
+    prop !== "isSmall",
 })<Pick<IconProps, "color" | "isDisabled" | "isSmall" | "isBig">>`
   width: ${({ isSmall, isBig }) => size(isSmall ? "smallIcon" : isBig ? "bigIcon" : "icon")};
   height: ${({ isSmall, isBig }) => size(isSmall ? "smallIcon" : isBig ? "bigIcon" : "icon")};

--- a/frontend/src/components/core/shared/input-fields/input-container/input-container.tsx
+++ b/frontend/src/components/core/shared/input-fields/input-container/input-container.tsx
@@ -27,7 +27,7 @@ const FlexContainer = styledDiv.div`
 `;
 
 const StyledInputFrame = styled.div.withConfig({
-  shouldForwardProp: (prop: string) => prop.toString() !== "smallBorder",
+  shouldForwardProp: (prop: string) => prop !== "smallBorder",
 })<{ smallBorder: boolean }>`
   box-sizing: border-box;
   background-color: ${color("transparent")};

--- a/frontend/src/models/sync-engine/rest-adapter.ts
+++ b/frontend/src/models/sync-engine/rest-adapter.ts
@@ -23,7 +23,7 @@ export class RESTAdapter<M> implements IStorageAdapter<M> {
 
   public async read<E extends keyof M>(entity: E, id: ID): Promise<Snapshot<M[E]> | undefined> {
     const response = await this.axiosInstance.get<Snapshot<M[E]>>(
-      `${this.baseUrl}/${this.mapEntityToRoute(entity)}/${String(id)}`,
+      `${this.baseUrl}/${this.mapEntityToRoute(entity)}/${id}`,
     );
     return response.data;
   }
@@ -44,14 +44,14 @@ export class RESTAdapter<M> implements IStorageAdapter<M> {
     newValue: Snapshot<Partial<M[E]>>,
   ): Promise<void> {
     await this.axiosInstance.patch(
-      `${this.baseUrl}/${this.mapEntityToRoute(entity)}/${String(id)}`,
+      `${this.baseUrl}/${this.mapEntityToRoute(entity)}/${id}`,
       newValue,
     );
   }
 
   public async delete<E extends keyof M>(entity: E, id: ID): Promise<void> {
     await this.axiosInstance.delete(
-      `${this.baseUrl}/${this.mapEntityToRoute(entity)}/${String(id)}`,
+      `${this.baseUrl}/${this.mapEntityToRoute(entity)}/${id}`,
     );
   }
 

--- a/frontend/src/utils/platform-detection.ts
+++ b/frontend/src/utils/platform-detection.ts
@@ -13,7 +13,7 @@ export const isFirefox = (): boolean =>
 
 export const isSafari = (): boolean =>
   navigator
-    ? Boolean(/^((?!chrome|android).)*safari/i.test(navigator.userAgent.toLowerCase()))
+    ? /^((?!chrome|android).)*safari/i.test(navigator.userAgent.toLowerCase())
     : false;
 
 export const isWindows = (): boolean =>

